modules/detect_target/detect_target.py

```diff
@@ -27,9 +27,12 @@ def __init__(self, device: "str | int", model_path: str, override_full: bool, sa
         self.__device = device
         self.__model = ultralytics.YOLO(model_path)
         self.__counter = 0
-        self.__enable_half_precision = False if self.__device == "cpu" else True
+        self.__enable_half_precision = False if self.__device == "cpu" else False
+        #modified so override_full controls if its half or full - FOR PROFILING ONLY 
```

Comment:
Instead of modifying the logic here, when profiling why not just specify the device type and the current logic will handle if half or full precision should be used?

Reply:
It might be worth updating this interface to a `create()` method that takes a settings object as argument. The settings class could have a `is_setting_valid()` method, which can push the responsibility of ensuring a valid setting to the caller. This also increases transparency because there won't be any fallthrough logic (the setting is applied as is).